File: backend/app/core/personas.py
"""User personas — stored in the existing `users.role` column. A persona shapes
the default experience (which tools are surfaced first, greeting, quick
actions); it is not a permission boundary except for `admin`, which cannot be
chosen at self-registration."""

PERSONAS = {
    "procurement_manager": {
        "label": "Procurement Manager",
        "description": "Decides when and how to buy freight: which origin, which vessel, spot or COA.",
        "focus": ["recommendation", "financial", "scenario"],
        "self_register": True,
    },
    "chartering_analyst": {
        "label": "Chartering Analyst",
        "description": "Watches the freight market and forecasts, and builds the case for a fixture.",
        "focus": ["markets", "forecast", "recommendation"],
        "self_register": True,
    },
    "port_ops": {
        "label": "Port & Logistics Officer",
        "description": "Manages berth fit, congestion and disruption risk at the East Coast ports.",
        "focus": ["ports", "risk", "map"],
        "self_register": True,
    },
    "finance_head": {
        "label": "Finance & Treasury",
        "description": "Owns freight cost exposure, hedging, and the savings case.",
        "focus": ["financial", "markets", "scenario"],
        "self_register": True,
    },
    "admin": {
        "label": "Administrator",
        "description": "Full access. Assigned by an existing administrator, not at sign-up.",
        "focus": ["markets", "forecast", "recommendation", "ports", "risk", "financial", "scenario"],
        "self_register": False,
    },
}

DEFAULT_PERSONA = "chartering_analyst"
SELF_REGISTER_PERSONAS = [k for k, v in PERSONAS.items() if v["self_register"]]

File: frontend/src/lib/personas.ts
import { Briefcase, LineChart, Anchor, Landmark, ShieldCheck, type LucideIcon } from "lucide-react";

export interface PersonaView {
  icon: LucideIcon;
  accent: string; // r,g,b for glow
  tagline: string;
  quickActions: { label: string; to: string; hint: string }[];
}

/** Presentation for each persona key returned by the backend. Unknown roles
 * (e.g. accounts created before personas existed) fall back to GENERIC. */
export const PERSONA_VIEW: Record<string, PersonaView> = {
  procurement_manager: {
    icon: Briefcase,
    accent: "14,116,144",
    tagline: "Decide when and how to buy freight.",
    quickActions: [
      { label: "Compare origins for a cargo", to: "/app/recommendation", hint: "Rank all 5 origins by cost and fit" },
      { label: "COA or spot?", to: "/app/financial", hint: "Simulate locking a contract vs staying spot" },
      { label: "Stress-test a decision", to: "/app/scenario", hint: "Freight spike, port closure, Red Sea" },
    ],
  },
  chartering_analyst: {
    icon: LineChart,
    accent: "109,40,217",
    tagline: "Read the market and build the case for a fixture.",
    quickActions: [
      { label: "Watch the live market", to: "/app/markets", hint: "Prices, spikes and regional boards" },
      { label: "Forecast a freight index", to: "/app/forecast", hint: "ARIMA + XGBoost with confidence bands" },
      { label: "Compare origins for a cargo", to: "/app/recommendation", hint: "Cost, transit and market direction" },
    ],
  },
  port_ops: {
    icon: Anchor,
    accent: "180,83,9",
    tagline: "Keep vessels matched to berths and ahead of disruption.",
    quickActions: [
      { label: "Check vessel-to-berth fit", to: "/app/ports", hint: "Draft, LOA, beam and tidal plans" },
      { label: "Score a route's risk", to: "/app/risk", hint: "Disruptions, congestion, volatility" },
      { label: "See the port map", to: "/app/map", hint: "Ports coloured by risk, vessels in transit" },
    ],
  },
  finance_head: {
    icon: Landmark,
    accent: "5,150,105",
    tagline: "Control freight cost exposure and prove the savings.",
    quickActions: [
      { label: "Size the savings", to: "/app/financial", hint: "ROI from better-timed chartering" },
      { label: "Watch INR and freight", to: "/app/markets", hint: "INR per USD alongside the freight indices" },
      { label: "Stress-test the budget", to: "/app/scenario", hint: "What a 50% freight spike does to cost" },
    ],
  },
  admin: {
    icon: ShieldCheck,
    accent: "220,38,38",
    tagline: "Full access across every tool.",
    quickActions: [
      { label: "Watch the live market", to: "/app/markets", hint: "Prices and regional boards" },
      { label: "Compare origins for a cargo", to: "/app/recommendation", hint: "Ranked comparison" },
      { label: "Score a route's risk", to: "/app/risk", hint: "Composite risk score" },
    ],
  },
};

export const GENERIC_VIEW: PersonaView = PERSONA_VIEW.chartering_analyst;

export const personaView = (role: string | undefined): PersonaView => PERSONA_VIEW[role ?? ""] ?? GENERIC_VIEW;

export const PERSONA_LABEL: Record<string, string> = {
  procurement_manager: "Procurement Manager",
  chartering_analyst: "Chartering Analyst",
  port_ops: "Port & Logistics Officer",
  finance_head: "Finance & Treasury",
  admin: "Administrator",
  analyst: "Analyst",
};
